### ISSUE TITLE
Idea: Trusted Domains in Container via Environment ?

### ISSUE THREAD

**Issue by HolgiB** [greyltc/docker-nextcloud#12]
Hi there,

thanks for your work ! The Nextcloud Container works very nicely. Usually Nextcloud configures itself to the LAN IP or the hostname of the Linux box it is running on. For the preconfigured container you provide only localhost is accepted as "trusted domain". Thus logins to the container via exposed ports to the docker host itself are blocked. Adding them to the "trusted domain" via button in the web browser didn't work for me. 

Do you think it would make sense to provide the Nextcloud image with a placeholder inside the config.php under Trusted Domains ? 

In this way people could set the Trusted Domains in the container during start by using Environment variables when starting it from the image.

What do you think ?

Regards,
Holger

**Comment by greyltc** [comment]
Hi Holger,
I've never had the problem of nextcloud blocking domains on me. Could you give me exact steps I should follow to recreate this problem?

**Comment by HolgiB** [comment]
Hi there,

strange, when I tried to access the container running on my other Xubuntu server via IP address or hostname I got an untrusted domain message from Nextcloud blocking the access. The same happened when I accessed the container from the internet via DynDNS Domain.

The steps to reproduce this is to access the Nextcloud container from an external system. Example: Running the container on another machine and accessing it from your Desktop via https://<ip-of-other-machine/nextcloud or https://hostname-of-other-machine/nextcloud. 

There seems to be an option to add the new untrusted "domain" from the browser to the Nextcloud trusted domains but this didn't work for me. Neither in Firefox, nor in Chrome.

I am running Xubuntu 16.04 x64 as desktop and Server OS if this is important.

**Comment by greyltc** [comment]
I think the very first time you visit your nextcloud site (when you get the initial request to setup your admin account and configure your database) nextcloud whitelists whatever domain you visit from (I'm pretty sure owncloud does this). Could it be that your initial visit and your subsequent visits come from different domains?

**Comment by HolgiB** [comment]
Hm, I am not shure about this. In the first place I did all initial setup via localhost and experimented from within my LAN (Nextcloud access via local IP) plus WAN (DynDNS) after it.

Possibly all works if you do the initial configuration from WAN. I will try this with a clean container in the first place and report back.

**Comment by greyltc** [comment]
I often deploy this container on servers geographically very far away from me over the internet and I've never had any trusted domain issues. I'm 99% sure the problem you're seeing is because you're setting up (& whitelisting) from localhost instead of doing your initial setup by visiting via the domain name you intend to use.

**Comment by HolgiB** [comment]
Yep, it seems that my only "fault" was calling the Docker container from localhost the first time. We set up a Nextcloud instance on a CentOS VM at our company and it behaved similar. When we called the Nextcloud web interface from the external FQDN, we could simply set it up without adding the domain to the "trusted domain". 

Sorry for taking your time ! This is not exactly obvious when one uses your NC container (or someone else container) in the first place ;-)
